src/plugins/explore/public/application/utils/state_management/actions/query_actions.ts

```diff
@@ -199,6 +206,44 @@ export const executeQueries = createAsyncThunk<
   }
 
   const defaultCacheKey = defaultPrepareQueryString(query);
+  // Use separate cache keys for data table and histogram
+  const dataTableCacheKey = prepareDataTableCacheKey(query);
+  const histogramCacheKey = defaultCacheKey;
```

Comment:
curious why histogram became the default cache key? I feel like histogram should have special handling instead instead of the data table